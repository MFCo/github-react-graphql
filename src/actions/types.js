import keyMirror from 'key-mirror';

export default keyMirror({
  UPDATE_USER: null,
  UPDATE_REPOSITORIES: null,
  ADD_REPOSITORIES: null,
  UPDATE_DIRECTORY: null,
  UPDATE_REPOSITORY: null
});