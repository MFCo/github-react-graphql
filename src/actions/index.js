import ACTIONS from './types';
import { createActions } from 'redux-actions';

export default createActions(...Object.keys(ACTIONS));