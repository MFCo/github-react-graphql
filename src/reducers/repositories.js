import { handleActions } from 'redux-actions';

const defaultState = {
  list: [],
  hasNextPage: false,
  endCursor: null
};

export default handleActions({
  UPDATE_REPOSITORIES: (state, { payload: { newList, newHasNextPage, newEndCursor } }) => {
    return {
      list: newList,
      hasNextPage: newHasNextPage,
      endCursor: newEndCursor
    }
  }
}, defaultState);