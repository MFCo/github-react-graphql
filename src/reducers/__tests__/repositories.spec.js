import repositories from '../repositories';
import actions from '../../actions';

it('should return the initial state', () => {
  expect(repositories(undefined, {})).toMatchSnapshot()
});

it('should handle UPDATE_REPOSITORIES', () => {
  expect(repositories(undefined,
    actions.addRepositories(
      {
        newList:
          [
            {
              "name": "PangRevenge",
              "description": "We should play",
              "forkCount": 0,
              "primaryLanguage": {
                "color": "#f1e05a",
                "name": "JavaScript"
              },
              "homepageUrl": null
            },
            {
              "name": "PangRevenge",
              "description": "Hahaha",
              "forkCount": 0,
              "primaryLanguage": {
                "color": "#f1e05a",
                "name": "JavaScript"
              },
              "homepageUrl": null
            }
          ],
        newHasNextPage: false,
        newEndCursor: "Y3Vyc29yOnYyOpHOARWdng=="
      }))).toMatchSnapshot()
});

it('should handle ADD_REPOSITORIES', () => {
  expect(repositories(
    {
      list:
        [
          {
            "name": "PangRevenge",
            "description": "Pang pang revenge",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          },
          {
            "name": "PangRevenge",
            "description": "Pangolier is balanced ",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          },
          {
            "name": "PangRevenge",
            "description": "Yeah, i love dota2 ",
            "forkCount": 0,
            "primaryLanguage": {
              "color": "#f1e05a",
              "name": "JavaScript"
            },
            "homepageUrl": null
          }
        ],
      hasNextPage: true,
      endCursor: "Y3Vyc29yOnYyOpHOARWdng=="
    },
    actions.addRepositories(
      {
        newList:
          {
            0: {
              "name": "PangRevenge",
              "description": "We should play",
              "forkCount": 0,
              "primaryLanguage": {
                "color": "#f1e05a",
                "name": "JavaScript"
              },
              "homepageUrl": null
            },
            1: {
              "name": "PangRevenge",
              "description": "Hahaha",
              "forkCount": 0,
              "primaryLanguage": {
                "color": "#f1e05a",
                "name": "JavaScript"
              },
              "homepageUrl": null
            }
          },
        newHasNextPage: false,
        newEndCursor: "Y3Vyc29yOnYyOpHOARWdng=="
      }))).toMatchSnapshot()
})