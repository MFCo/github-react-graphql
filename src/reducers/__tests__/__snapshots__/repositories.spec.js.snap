// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle ADD_REPOSITORIES 1`] = `
Object {
  "endCursor": "\\"Y3Vyc29yOnYyOpHOARWdng==\\"",
  "hasNextPage": false,
  "list": Array [
    Object {
      "description": "Pang pang revenge",
      "homepageUrl": null,
      "name": "PangRevenge",
      "primaryLanguage": Object {
        "color": "#f1e05a",
        "name": "JavaScript",
      },
      "viewerHasStarred": false,
      "viewerSubscription": "UNSUBSCRIBED",
    },
    Object {
      "description": "Pangolier is balanced ",
      "homepageUrl": null,
      "name": "PangRevenge",
      "primaryLanguage": Object {
        "color": "#f1e05a",
        "name": "JavaScript",
      },
      "viewerHasStarred": false,
      "viewerSubscription": "SUBSCRIBED",
    },
    Object {
      "description": "Yeah, i love dota2 ",
      "forkCount": 0,
      "homepageUrl": null,
      "name": "PangRevenge",
      "primaryLanguage": Object {
        "color": "#f1e05a",
        "name": "JavaScript",
      },
    },
    Object {
      "0": Object {
        "description": "We should play",
        "homepageUrl": null,
        "name": "PangRevenge",
        "primaryLanguage": Object {
          "color": "#f1e05a",
          "name": "JavaScript",
        },
        "viewerHasStarred": false,
        "viewerSubscription": "SUBSCRIBED",
      },
      "1": Object {
        "description": "Hahaha",
        "forkCount": 0,
        "homepageUrl": null,
        "name": "PangRevenge",
        "primaryLanguage": Object {
          "color": "#f1e05a",
          "name": "JavaScript",
        },
      },
    },
  ],
}
`;

exports[`should handle UPDATE_REPOSITORIES 1`] = `
Object {
  "endCursor": "\\"Y3Vyc29yOnYyOpHOARWdng==\\"",
  "hasNextPage": false,
  "list": Array [
    Object {
      "description": "We should play",
      "homepageUrl": null,
      "name": "PangRevenge",
      "primaryLanguage": Object {
        "color": "#f1e05a",
        "name": "JavaScript",
      },
      "viewerHasStarred": false,
      "viewerSubscription": "UNSUBSCRIBED",
    },
    Object {
      "description": "Hahaha",
      "homepageUrl": null,
      "name": "PangRevenge",
      "primaryLanguage": Object {
        "color": "#f1e05a",
        "name": "JavaScript",
      },
      "viewerHasStarred": false,
      "viewerSubscription": "UNSUBSCRIBED",
    },
  ],
}
`;

exports[`should return the initial state 1`] = `
Object {
  "endCursor": null,
  "hasNextPage": false,
  "list": Array [],
}
`;
